File: examples/docs-code-snippets/src/main/scala/examples/scala/interactions/ReserveItems.scala
package examples.scala.interactions

import com.ing.baker.recipe.scaladsl.{Ingredient, Interaction}

trait ReserveItems extends Interaction {

  sealed trait Outcome

  case class OrderHasUnavailableItems(unavailableProductIds: List[String]) extends Outcome

  object Success extends Outcome

  val reserve: Interaction = Interaction(
    name = "ReserveItems",
    inputIngredients = Seq(
      Ingredient[String](name = "orderId"),
      Ingredient[List[String]](name = "productIds")
    ),
    output = Seq(

    )
  )

}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/interactions/ReserveItems.kt
package examples.kotlin.interactions

import com.ing.baker.recipe.javadsl.Interaction

interface ReserveItems : Interaction {
    sealed interface Outcome

    data class OrderHasUnavailableItems(val unavailableProductIds: List<String>) : Outcome
    object Success : Outcome

    fun apply(orderId: String, productIds: List<String>): Outcome
}
